### PR TITLE
python3Packages.openshift: init at 0.12.0

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change

I needed openshift for the official kubernetes ansible module.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119960` at 735d3144 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.openshift</li>
    <li>python38Packages.python-string-utils</li>
    <li>python39Packages.openshift</li>
    <li>python39Packages.python-string-utils</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/openshift/default.nix:32:0:
```
   |
32 |     description = "Python client for the OpenShift API";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/python-string-utils/default.nix:17:0:
```
   |
17 |     description = "A handy Python library to validate, manipulate and generate strings.";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/openshift/default.nix:35:5:
```
   |
35 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/openshift/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  buildInputs = [
+    jinja2
+    kubernetes
+    python-string-utils
+    ruamel-yaml
+    six
+  ];
+
+  doCheck = false;
```
Comment (lukegb) [inline pkgs/development/python-modules/openshift/default.nix:29]: Why? Do the tests not pass? (ref [§15.20.1.2.3](https://nixos.org/manual/nixpkgs/unstable/#testing-python-packages)) Are they not shipped in the Pypi build?

Please leave a comment, at least, and consider using a pythonImportsCheck if disabling tests cannot be avoided.

File: pkgs/development/python-modules/openshift/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  lib
+  , buildPythonPackage
+  , fetchPypi
+  , jinja2
+  , kubernetes
+  , ruamel-yaml
+  , six
+  , python-string-utils
```
Comment (lukegb) [inline pkgs/development/python-modules/openshift/default.nix:9]: This formatting is a bit non-standard. I wouldn't usually call this out, but this doesn't fit any of the patterns I've seen used commonly.

File: pkgs/development/python-modules/openshift/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openshift/default.nix:10]:
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, jinja2
, kubernetes
, ruamel-yaml
, six
, python-string-utils
}:
```

File: pkgs/development/python-modules/openshift/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  meta = with lib; {
+    description = "Python client for the OpenShift API";
+    homepage = "https://github.com/openshift/openshift-restclient-python";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
+  };
+}
+
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openshift/default.nix:39]: To many final new lines.

File: pkgs/development/python-modules/openshift/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/openshift/default.nix:29]: > pythonImportsCheck

Should always be used to make sure check only inputs are not required to import the package.

File: pkgs/development/python-modules/openshift/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+buildPythonPackage rec {
+  pname = "openshift";
+  version = "0.12.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-aggRnD4goiZJPp4cngp8AIrJC/V46378cwUSfq8Xml4=";
+  };
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openshift/default.nix:21]:
```suggestion
  propagatedBuildInputs = [
```

File: pkgs/development/python-modules/openshift/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    six
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Python client for the OpenShift API";
+    homepage = "https://github.com/openshift/openshift-restclient-python";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openshift/default.nix:35]: Missing maintainer.

File: pkgs/development/python-modules/python-string-utils/default.nix [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{
+  lib
+  , buildPythonPackage
+  , fetchPypi
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-string-utils/default.nix:5]:
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
}:
```

File: pkgs/development/python-modules/python-string-utils/default.nix [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+buildPythonPackage rec {
+  pname = "python-string-utils";
+  version = "1.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-3PkGCwPwdkfApgNAjciwP4B/O1SgXG4Z6xRGAlb6wMs=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-string-utils/default.nix:14]: Missing checkPhase and pythonImportsCheck.

File: pkgs/development/python-modules/openshift/default.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/python-modules/openshift/default.nix:29]: didnt know about pythonImportsCheck, I've wanted that for a long time cool ! thanks

File: pkgs/development/python-modules/python-string-utils/default.nix (hunk 8 above)
Comment (teto) [reply pkgs/development/python-modules/python-string-utils/default.nix:14]: tests are nbot bundled via pypi, I added a doCheck = false; with a comment to explain this. I've added the import check

**Comment by teto** [comment]
dunno if it is transient but seems like there is an issue with fetchPypi / pypi
```
curl -L https://pypi.io/packages/source/o/openshift/openshift-0.12.0.tar.gz                                                                                                                                                                            ~/home
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.13.9</center>
</body>
</html>
```

File: pkgs/development/python-modules/python-string-utils/default.nix [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+  checkInputs = [ pytest ];
```
Comment (lukegb) [inline pkgs/development/python-modules/python-string-utils/default.nix:16]: Hmm, if you're not running the tests, you shouldn't need pytest here.

File: pkgs/development/python-modules/python-string-utils/default.nix [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  pythonImportsCheck = ["string_utils"];
+
+  # tests are not available from pypi
+  doCheck=false;
```
Comment (lukegb) [inline pkgs/development/python-modules/python-string-utils/default.nix:20]:
```suggestion
  doCheck = false;
```

**Review by lukegb — COMMENTED** [review]
Some minor nits: if you tidy them up and fix the merge conflict I'll merge.

**Comment by teto** [comment]
done

**Comment by lukegb** [comment]
@ofborg build python3.pkgs.openshift

**Pull request merged**